pilot/pkg/proxy/envoy/externalservice.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envoy
+
+import (
+	"fmt"
+	// TODO(nmittler): Remove this
+	_ "github.com/golang/glog"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	routingv2 "istio.io/api/routing/v1alpha2"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+func buildExternalServicePort(port *routingv2.Port) *model.Port {
+	protocol := model.ConvertCaseInsensitiveStringToProtocol(port.Protocol)
+	return &model.Port{
+		Name:     fmt.Sprintf("external-%v-%d", protocol, port.Number), // TODO: use external service port name in building model port name?
```

Comment:
Can you file an issue and reference it here, this is definitely going to trip people up and we'll want to revisit,